### PR TITLE
[enhance](BE) Refactor TaskWorkerPool

### PR DESCRIPTION
**Pull request by platoneko**
## Proposed changes

Refactor TaskWorkerPool

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -324,1709 +145,1552 @@ void TaskWorkerPool::_finish_task(const TFinishTaskRequest& finish_task_request)
...
-                    worker_thread_lock, [this]() { return !_is_work || !_tasks.empty(); });
-            if (!_is_work) {
-                return;
-            }
-
-            agent_task_req = _tasks.front();
-            upload_request = agent_task_req.upload_req;
-            _tasks.pop_front();
-        }
+bvar::Adder<uint64_t> report_task_total("report", "task_total");
```
Comment (gavinchou) [inline be/src/agent/task_worker_pool.cpp:396]: ditto

**Review by gavinchou — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**